manual/opt.tex

```diff
@@ -153,9 +153,108 @@ \subsubsection{quartic}
+as ``stablizers''. The generalized eigenvalue problem is solved for each shifted Hamiltonian and updates to wave function parameters are obtainted. Then a correlated sampling 
+is performed (using the middle shift's updated wave function as the guiding function) for each shift's updated wave function and the initial trial wave function.
+The cost function for these wave functions is compared, and the update corresponding to the best cost function is selected.
+In the next iteration, the median magnitude of the stabilizers is set to that that generated the best update in the current iteration, thus adapting the magnitude of
+the stabilizers automatically.
+
+Thanks to recently developments in excited state variational principles, the adaptive optimizer is also able to optimize individual excited states directly.  
+In this case, it tries to minimize the following function: 
+\begin{equation*}
+\Omega[\Psi]=\frac{\left<\Psi|\omega-H|\Psi\right>}{\left<\Psi|{\left(\omega-H\right)}^2|\Psi\right>}
+\end{equation*}
+The global minimal of this functional corresponds to the state whose energy lies immediately above the shift parameter $\omega$ in the energy spectrum.
+For example, suppose the initial 
+trial wave function's parameters are taken as a guess for the first excited state.
+Then, by setting $\omega$ to be a value in between the ground state and first excited state 
+energies would cause the trial wave function to be optimized for the first excited state. 
+
+When the trial wave function contains more than tens of thousands of parameters, constructing and storing the linear method matrices may become a memory bottleneck. 
+To avoid explicit
+construction of these matrices, the adaptive optimizer implements the recently developed ``block linear method''.  The block linear method (BLM) tries to find an approximate 
+solution $\vec{c}_{opt}$ to the standard LM generalized eigenvalue problem by (a) dividing the variable space into a number of blocks (b) making intelligent estimates for which 
+directions within those blocks will be most important for constructing $\vec{c}_{opt}$, and (c) estimating $\vec{c}_{opt}$ by solving a smaller, more memory-efficient 
+eigenproblem in the basis of these supposedly important block-wise directions. 
+
+\begin{table}[h]
+\begin{center}
+\begin{tabularx}{\textwidth}{l l l l l l }
+\hline
+\multicolumn{6}{l}{\texttt{linear} method} \\
+\hline
+\multicolumn{2}{l}{parameters}  & \multicolumn{4}{l}{}\\
+   &   \bfseries name     & \bfseries datatype & \bfseries values & \bfseries default   & \bfseries description \\
+   %&   \texttt{stepsize} &  real     & 0--1 & 0.25  & Step size for moving parameters\\
+   &   \texttt{max\_relative\_change} &  real     & $>0$ & 10.0 & Allowed change in cost function\\
+   &   \texttt{max\_param\_change} &  real     & $>0$ & 0.3 & Allowed change in wave function parameter\\
+   &   \texttt{shift\_i} &  real     & $>0$ & 0.01 & Direct stabilizer added to the Hamiltonian matrix\\
```

Comment:
Change "Direct" to "Initial". OSO doesn't update shift_i but adaptive does. So the user input is an initial value.